janusgraph-backend-testutils/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java

```diff
@@ -1205,15 +1205,23 @@ public void clearStorageTest() throws Exception {
     @MethodSource("org.janusgraph.core.attribute.TextArgument#text")
     public void testTextPredicate(JanusGraphPredicate predicate, boolean expected, String value, String condition) throws BackendException {
         assumeIndexSupportFor(Mapping.TEXT, predicate);
-        initializeWithDoc("vertex", "test1", TEXT, value, true);
+        if (value != null)
+            initializeWithDoc("vertex", "test1", TEXT, value, true);
+        else
+            // if the value is null, replicate a missing field by indexing a different one
+            initializeWithDoc("vertex", "test1", BOOLEAN, true, true);
         testPredicateByCount((expected) ? 1 : 0, predicate, TEXT, condition);
     }
 
     @ParameterizedTest
     @MethodSource("org.janusgraph.core.attribute.TextArgument#string")
     public void testStringPredicate(JanusGraphPredicate predicate, boolean expected, String value, String condition) throws BackendException {
         assumeIndexSupportFor(Mapping.STRING, predicate);
-        initializeWithDoc("vertex", "test1", NAME, value, true);
+        if (value != null)
+            initializeWithDoc("vertex", "test1", NAME, value, true);
+        else
+            // if the value is null, replicate a missing field by indexing a different one
+            initializeWithDoc("vertex", "test1", BOOLEAN, true, true);
```

Comment:
I dont like this workaround, but I couldnt find a simpler way to insert an arbitrary document without any properties. Up for any suggestions on making this cleaner.

Reply:
This looks fine to me. Anyway this is a test so I wouldn't be picky :)